tasks/Task.cpp

```diff
@@ -235,3 +285,27 @@ void Task::cleanupHook()
 {
     TaskBase::cleanupHook();
 }
+
+std::vector<uint8_t> parseManagedDigitalOutputs(
+    linux_gpios::WritePathConfiguration const& config)
+{
+    std::string pattern("[Dd](\\d{2})$");
+    std::regex regex(pattern);
+
+    std::vector<uint8_t> output_id;
+    output_id.reserve(config.gpio_paths.size());
+
+    for (std::string const& path : config.gpio_paths) {
+        std::smatch id;
+        if (!std::regex_match(path, id, regex)) {
+            throw std::runtime_error(path +
+                                     "is not valid a valid digital output. "
+                                     "It does not match " +
+                                     pattern);
+        }
+
+        output_id.push_back(std::stoi(id[1].str()));
+    }
+
+    return output_id;
```

Comment:
As we discussed, I think that just having a number in the path would make things a lot simpler.